src/main/java/org/cryptomator/ui/mainwindow/ResizeController.java

```diff
@@ -40,61 +40,56 @@ public class ResizeController implements FxController {
-				window.setY((primaryScreenBounds.getHeight() - window.getMinHeight()) / 2);
-				window.setWidth(window.getMinWidth());
-				window.setHeight(window.getMinHeight());
-			} else {
-				window.setHeight(settings.windowHeight.get() > window.getMinHeight() ? settings.windowHeight.get() : window.getMinHeight());
-				window.setWidth(settings.windowWidth.get() > window.getMinWidth() ? settings.windowWidth.get() : window.getMinWidth());
-				window.setX(settings.windowXPosition.get());
-				window.setY(settings.windowYPosition.get());
-			}
+		if (!neverTouched()) {
+			window.setHeight(settings.windowHeight.get() > window.getMinHeight() ? settings.windowHeight.get() : window.getMinHeight());
+			window.setWidth(settings.windowWidth.get() > window.getMinWidth() ? settings.windowWidth.get() : window.getMinWidth());
+			window.setX(settings.windowXPosition.get());
+			window.setY(settings.windowYPosition.get());
 		}
-		savePositionalSettings();
+		window.setOnShowing(this::checkDisplayBounds);
 	}
 
 	private boolean neverTouched() {
 		return (settings.windowHeight.get() == 0) && (settings.windowWidth.get() == 0) && (settings.windowXPosition.get() == 0) && (settings.windowYPosition.get() == 0);
 	}
 
-	private boolean didDisplayConfigurationChange() {
-		String currentDisplayConfiguration = getMonitorSizes();
-		String settingsDisplayConfiguration = settings.displayConfiguration.get();
-		boolean configurationHasChanged = !settingsDisplayConfiguration.equals(currentDisplayConfiguration);
-		if (configurationHasChanged) settings.displayConfiguration.set(currentDisplayConfiguration);
-		return configurationHasChanged;
-	}
-
-	private String getMonitorSizes() {
-		ObservableList<Screen> screens = Screen.getScreens();
-		StringBuilder sb = new StringBuilder();
-		for (int i = 0; i < screens.size(); i++) {
-			Rectangle2D screenBounds = screens.get(i).getBounds();
-			if (!sb.isEmpty()) sb.append(" ");
-			sb.append("displayId: " + i + ", " + screenBounds.getWidth() + "x" + screenBounds.getHeight() + ";");
+	private void checkDisplayBounds(WindowEvent evt) {
+		// Rect of half the height of the title bar
+		Rectangle2D titleBarRect = new Rectangle2D(window.getX(), window.getY(), window.getWidth(), titleBar.getMinHeight() / 2);
+
+		// Fraction f defines the minimum amount of the width/height that must be visible to not reset the window. Must be less or equal to 0.5
+		double f = 0.5;
+		double minX = window.getX() + f * window.getWidth();
+		double minY = window.getY() + f * window.getHeight();
+		double width = window.getWidth() - 2 * f * window.getWidth();
+		double height = window.getHeight() - 2 * f * window.getHeight();
+		Rectangle2D centerRect = new Rectangle2D(minX, minY, width, height);
+
+		// Half the title bar or the center rect is out of display bounds
+		if (Screen.getScreensForRectangle(titleBarRect).isEmpty() || Screen.getScreensForRectangle(centerRect).isEmpty()) {
```

Comment:
With this logic, the following edge case behaves... not entirely as expected?
 Consider a system with the system bar _not_ on the bottom. If i drag the Cryptomator window to the bottom, as soon as i can only see the title bar, it will be reset to primary center on minimize and maximize.
 
 @tobihagemann This okay for you?

Reply:
Apparently it is not possible to move the system bar anymore in Windows 11 so I can't really reproduce. However, in the version you tested the threshold was set to 50%. So if you move the window to the bottom of the screen, such that only the title bar is visible, then less than 50% is visible so it was reset.

> it will be reset to primary center on minimize and maximize.

You mean minimize to system bar or to tray?
And by maximize you mean fullscreen or restore from minimize?

Reply:
This was a logical conclusion, i did not tested it with your branch. The condition says, if either one of the two rectangles (titlebar and main app) is not inside a screen, execute the following bracket scope

> You mean minimize to system bar or to tray?

Should not make a difference, since we added a listener to the `Window.onShowingProperty`. With maximize i meant "show the window again" 😉